### ISSUE TITLE
Training loss and generalization during test

### ISSUE THREAD

**Issue by hit2sjtu** [amaralibey/MixVPR#15]
I tried tuning some training params and sometimes the training loss is lower than using the default params in the paper but generalizes worse on our own data.  it seems like overfitting but I am not sure.  Besides, the git history shows that you have tried using dropout. Does it help in your experiments?

**Comment by amaralibey** [comment]
Hello @hit2sjtu,

The loss function is not always a reliable indicator of performance, especially in metric learning. Its value is depends  entirely on the relationship between pairs in each mini-batch, which is different from classification where each image is classified into one class (among all the existing classes) independently of the batch-size or the images present in the same mini-batch. 

For example, changing the augmentation operations can have a significant impact on the minimum achievable loss value (more data augmentation -> harder mini-batches (which translates to higher loss values) -> more pairwise information to squeez -> better representation learning). 

If you only changed the learning rate and optimizer parameters while using the same loss function (with the same hyperparameters that control it), then hitting a local optima could be a possible explanation for the observed behavior. 

In practice, it's better to rely on the validation performance, rather than the loss value.

Finally, the dropout operation had no performance effect in our experiments.

**Comment by hit2sjtu** [comment]
Thanks for your clarification, I guess the metric learning was the issue. During my test, the memory has large impact on the final results, I guess this is also due to more pairs in each mini-batch.

**Comment by hit2sjtu** [comment]
glad to know. I will check your new paper.
